theseus/src/state/projects.rs

```diff
@@ -138,6 +160,44 @@ pub struct Dependency {
     pub dependency_type: DependencyType,
 }
 
+impl Dependency {
+    // As Modrinth doesn't always provide the project ID, we have to infer it from the version ID
+    pub async fn populate(&mut self) -> crate::Result<()> {
+        let mut dep = self;
```

Comment:
this IMO is a pretty bad way of solving this specific issue. I think you can remove the fix for this GH issue and I can fix this one